### PR TITLE
Add missing CultureInfo.InvariantCulture

### PR DESCRIPTION
**Pull request by latop2604**
This pull request cover a subset of pages.
I also fix an issue `GraphController.Spark.cs` due to `Enumerable.Max` (with not nullable type) that raise exception if the collection is empty.

### CODE REVIEW

File: Opserver/Controllers/GraphController.Spark.cs [hunk 1]
```diff
@@ -92,18 +93,18 @@ public ActionResult SQLCPUSpark(string node)
...
             {
-                sb.Append((p.DateEpoch - startEpoch) / range).Append(" ")
-                  .Append((height - getVal(p) / divisor).ToString("n1")).Append(" ");
+                sb.Append(((p.DateEpoch - startEpoch) / range).ToString(CultureInfo.InvariantCulture)).Append(" ")
+                  .Append((height - getVal(p) / divisor).ToString("n1", CultureInfo.InvariantCulture)).Append(" ");
             }
-            sb.AppendFormat(@"{0} {1} z""/>
+            sb.AppendFormat(CultureInfo.InvariantCulture, @"{0} {1} z""/>
    </g>
-</svg>", width.ToString(), height.ToString());
```
Comment (NickCraver) [inline Opserver/Controllers/GraphController.Spark.cs]: Why were these removed?

File: Opserver/Controllers/GraphController.Spark.cs [hunk 2]
```diff
@@ -63,10 +64,10 @@ public partial class GraphController
             if (iface == null) return ContentNotFound();
             var points = await iface.GetUtilization(SparkStart, null, SparkPoints);
 
-            Func<DoubleGraphPoint, double> getter = p => p.Value.GetValueOrDefault(0);
-            if (direction == "out") getter = p => p.BottomValue.GetValueOrDefault(0);
+            Func<DoubleGraphPoint, double?> getter = p => p.Value;
+            if (direction == "out") getter = p => p.BottomValue;
 
-            return SparkSVG(points, Convert.ToInt64(points.Max(getter)), getter);
+            return SparkSVG(points, Convert.ToInt64(points.Max(getter)), p => getter(p).GetValueOrDefault(0));
```
Comment (NickCraver) [inline Opserver/Controllers/GraphController.Spark.cs]: This removes the null protection for the max accessor.

File: Opserver/Controllers/GraphController.Spark.cs [hunk 3]
```diff
@@ -92,18 +93,18 @@ public ActionResult SQLCPUSpark(string node)
                  divisor = max/50,
                  range = (nowEpoch - startEpoch)/width;
 
-            var sb = StringBuilderCache.Get().AppendFormat(@"<svg version=""1.1"" baseProfile=""full"" width=""{0}"" height=""{1}"" xmlns=""http://www.w3.org/2000/svg"" preserveAspectRatio=""none"">
+            var sb = StringBuilderCache.Get().AppendFormat(CultureInfo.InvariantCulture, @"<svg version=""1.1"" baseProfile=""full"" width=""{0}"" height=""{1}"" xmlns=""http://www.w3.org/2000/svg"" preserveAspectRatio=""none"">
   <line x1=""0"" y1=""{1}"" x2=""{0}"" y2=""{1}"" stroke=""{3}"" stroke-width=""1"" />
   <g fill=""{2}"" stroke=""none"">
-    <path d=""M0 50 L", width.ToString(), height.ToString(), Color, AxisColor);
```
Comment (NickCraver) [inline Opserver/Controllers/GraphController.Spark.cs]: Same as below, why were the `.ToString()` calls removed?

File: Opserver/Views/Dashboard/Dashboard.Table.cshtml [hunk 4]
```diff
@@ -87,7 +88,7 @@ else
                                     @if (n.MemoryUsed >= 0)
                                     {
                                         <div class="progress" style="width: 215px;">
-                                            <div class="progress-bar @n.MemoryMonitorStatus().ProgressBarClass()" style="width: @(n.PercentMemoryUsed.ToString())%"></div>
+                                            <div class="progress-bar @n.MemoryMonitorStatus().ProgressBarClass()" style="width: @(n.PercentMemoryUsed.Value.ToString(CultureInfo.InvariantCulture))%"></div>
```
Comment (NickCraver) [inline Opserver/Views/Dashboard/Dashboard.Table.cshtml]: Introduces a null ref

File: Opserver/Views/Dashboard/Node.cshtml [hunk 5]
```diff
@@ -73,14 +74,14 @@
             <div class="panel-body">
                 <div class="value-block small col-xs-6" title="@s.ApplicationCPUTextSummary()">
                     <div class="progress">
-                        <div class="progress-bar" style="width: @s.CPULoad.ToString()%;"></div>
+                        <div class="progress-bar" style="width: @s.CPULoad.Value.ToString(CultureInfo.InvariantCulture)%;"></div>
```
Comment (NickCraver) [inline Opserver/Views/Dashboard/Node.cshtml]: Introduces a null ref

File: Opserver/Views/Dashboard/Node.cshtml [hunk 6]
```diff
@@ -73,14 +74,14 @@
...
                         <span>@s.CPUStatusSpan()</span>
                     </div>
                     <label>CPU</label>
                 </div>
                 <div class="value-block small col-xs-6" title="@s.ApplicationMemoryTextSummary()">
                     <div class="progress">
-                        <div class="progress-bar @s.MemoryMonitorStatus().ProgressBarClass()" style="width: @s.PercentMemoryUsed.ToString()%;"></div>
+                        <div class="progress-bar @s.MemoryMonitorStatus().ProgressBarClass()" style="width: @s.PercentMemoryUsed.Value.ToString(CultureInfo.InvariantCulture)%;"></div>
```
Comment (NickCraver) [inline Opserver/Views/Dashboard/Node.cshtml]: Introduces a null ref

File: Opserver/Views/Dashboard/Node.cshtml [hunk 7]
```diff
@@ -121,7 +122,7 @@
                     {
                         <div class="value-block small col-xs-12" title="@v.Description: @(v.Used?.ToString("n0")) of @(v.Size?.ToString("n0")) bytes used">
                             <div class="progress">
-                                <div class="progress-bar @(v.SpaceStatus.ProgressBarClass())" style="width: @(v.PercentUsed.ToString())%;"></div>
+                                <div class="progress-bar @(v.SpaceStatus.ProgressBarClass())" style="width: @(v.PercentUsed.Value.ToString(CultureInfo.InvariantCulture))%;"></div>
```
Comment (NickCraver) [inline Opserver/Views/Dashboard/Node.cshtml]: Introduces a null ref

File: Opserver/Views/Hub/Dashboard.Node.cshtml [hunk 8]
```diff
@@ -40,7 +41,7 @@
     <div class="status-block mem-status">
         <span class="label">Mem </span>@s.MemoryPercentStatusSpan()
         <div class="status-graph">@s.MemoryStatusSpan()@if (s.HasValidMemoryReading) {
-            <div class="space-used" style="width: 215px;"><div class="used @s.MemoryMonitorStatus().GetDescription()" style="width:@(s.PercentMemoryUsed)%"></div></div>
+            <div class="space-used" style="width: 215px;"><div class="used @s.MemoryMonitorStatus().GetDescription()" style="width: @s.PercentMemoryUsed.Value.ToString(CultureInfo.InvariantCulture)%"></div></div>
```
Comment (NickCraver) [inline Opserver/Views/Hub/Dashboard.Node.cshtml]: Introduces a null ref

File: Opserver/Views/SQL/Instance.cshtml [hunk 9]
```diff
@@ -159,7 +159,7 @@ if (counter == null)
                         <div class="panel-body small">
                             <div class="value-block col-md-3">
                                 <div class="progress">
-                                    <div class="progress-bar" style="width: @i.CurrentCPUPercent.ToString()%;"></div>
+                                    <div class="progress-bar" style="width: @i.CurrentCPUPercent.Value.ToString()%;"></div>
```
Comment (NickCraver) [inline Opserver/Views/SQL/Instance.cshtml]: Introduces a null ref

File: Opserver/Views/SQL/Instance.cshtml [hunk 10]
```diff
@@ -190,7 +190,7 @@ Top Users:
     <text>  @mc.Name: @mc.UsedBytes.ToSize() (@(mc.UsedPercent.ToString("#0.##"))%)</text>
 }">
                                 <div class="progress">
-                                    <div class="progress-bar" style="width: @(i.CurrentMemoryPercent.ToString())%;"></div>
+                                    <div class="progress-bar" style="width: @(i.CurrentMemoryPercent.Value.ToString(CultureInfo.InvariantCulture))%;"></div>
```
Comment (NickCraver) [inline Opserver/Views/SQL/Instance.cshtml]: Introduces a null ref

**Comment by NickCraver** [comment]
I have a changeset for these locally, but have been very busy at Stack the past few weeks. I don't want to be harsh here because that's really not my goal in *any* pull request - but it's probably better if we close this out and I commit a version with all of the proper null-protected changes that also doesn't introduce boxing.

File: Opserver/Controllers/GraphController.Spark.cs (hunk 2 above)
Comment (latop2604) [inline Opserver/Controllers/GraphController.Spark.cs]: Arf this is a last minute change. I initially use `? :` around `Convert.ToInt64` call .

File: Opserver/Controllers/GraphController.Spark.cs (hunk 1 above)
Comment (latop2604) [inline Opserver/Controllers/GraphController.Spark.cs]: To be handle by the culture info in AppendFormat.

File: Opserver/Views/Dashboard/Dashboard.Table.cshtml (hunk 4 above)
Comment (latop2604) [inline Opserver/Views/Dashboard/Dashboard.Table.cshtml]: Yes I see that. I assume that the previous `@if (n.MemoryUsed >= 0)` was enough.

**Comment by latop2604** [comment]
Ok, sorry for your time.

File: Opserver/Controllers/GraphController.Spark.cs (hunk 1 above)
Comment (NickCraver) [inline Opserver/Controllers/GraphController.Spark.cs]: For the integer case specifically, the only thing that can change for integers (the only thing involves in the `.AppendFormat()` cases) is the negative sign, which isn't in play. So we can leave those `.ToString()` to eliminate boxing by their original calls.

File: Opserver/Views/Dashboard/Dashboard.Table.cshtml (hunk 4 above)
Comment (NickCraver) [inline Opserver/Views/Dashboard/Dashboard.Table.cshtml]: I should have clarified here: `TotalMemory` can be null in some cases, that's where this would throw - fixing in a commit via `?.` approach instead (same for many others).

**Comment by NickCraver** [comment]
@latop2604 I really wanted your name in the commit log here after spending time on this - I've just merged in and just added the few fixes from items in comments as follow-ups. I'd really appreciate it if you could test the updated code and see if you're still running into any issues.

**Comment by latop2604** [comment]
It seems to be good on my side. Thx for your time and your feedback too. You're awesome ;)
